src/tlo/methods/cervical_cancer.py

```diff
@@ -0,0 +1,1515 @@
+        super().__init__(module, frequency=DateOffset(months=1))
+        # scheduled to run every 1 month: do not change as this is hard-wired into the values of all the parameters.
+
+    def apply(self, population):
+        df = population.props  # shortcut to dataframe
+        m = self.module
+        rng = m.rng
+        p = self.sim.modules['CervicalCancer'].parameters
+
+        # ------------------- SET INITIAL CE_HPV_CC_STATUS -------------------------------------------------------------------
+        # this was done here and not at outset because baseline value of hv_inf was not accessible
+
+        given_date = pd.to_datetime('2010-02-03')
+
+        if self.sim.date < given_date:
+
+            women_over_15_nhiv_idx = df.index[(df["age_years"] > 15) & (df["sex"] == 'F') & ~df["hv_inf"]]
+
+            df.loc[women_over_15_nhiv_idx, 'ce_hpv_cc_status'] = rng.choice(
+                ['none', 'hpv', 'cin1', 'cin2', 'cin3', 'stage1', 'stage2a', 'stage2b', 'stage3', 'stage4'],
+                size=len(women_over_15_nhiv_idx), p=p['init_prev_cin_hpv_cc_stage_nhiv']
+            )
+
+            women_over_15_hiv_idx = df.index[(df["age_years"] > 15) & (df["sex"] == 'F') & df["hv_inf"]]
+
+            df.loc[women_over_15_hiv_idx, 'ce_hpv_cc_status'] = rng.choice(
+                ['none', 'hpv', 'cin1', 'cin2', 'cin3', 'stage1', 'stage2a', 'stage2b', 'stage3', 'stage4'],
+                size=len(women_over_15_hiv_idx), p=p['init_prev_cin_hpv_cc_stage_hiv']
+            )
+
+        # -------------------- ACQUISITION AND PROGRESSION OF CANCER (ce_hpv_cc_status) -----------------------------------
+
+        df.ce_new_stage_this_month = False
+
+        df['ce_hiv_unsuppressed'] = ((df['hv_art'] == 'on_not_vl_suppressed') | (df['hv_art'] == 'not')) & (df['hv_inf'])
+
+        # determine if the person had a treatment during this stage of cancer (nb. treatment only has an effect on
+        #  reducing progression risk during the stage at which is received.
+
+        for stage, lm in self.module.linear_models_for_progression_of_hpv_cc_status.items():
+            gets_new_stage = lm.predict(df.loc[df.is_alive], rng)
+
+            idx_gets_new_stage = gets_new_stage[gets_new_stage].index
+
+#           print(stage, lm, gets_new_stage, idx_gets_new_stage)
+
+            df.loc[idx_gets_new_stage, 'ce_hpv_cc_status'] = stage
+            df.loc[idx_gets_new_stage, 'ce_new_stage_this_month'] = True
+
+        df['ce_cc_ever'] = ((df.ce_hpv_cc_status == 'stage1') | (df.ce_hpv_cc_status == 'stage2a')
+                            | (df.ce_hpv_cc_status == 'stage2b') | (df.ce_hpv_cc_status == 'stage3') | (
+                                    df.ce_hpv_cc_status == 'stage4')
+                            | df.ce_ever_treated)
+
+        # -------------------------------- SCREENING FOR CERVICAL CANCER USING XPERT HPV TESTING AND VIA---------------
+        # A subset of women aged 30-50 will receive a screening test
```

Comment:
should the boundaries be included(30yrs and 50yrs) selection on Ln 720 is excluding them